activesupport/lib/active_support/cache.rb

```diff
@@ -333,19 +333,16 @@ def read_multi(*names)
         options = merged_options(options)
 
         instrument_multi(:read, names, options) do |payload|
-          results = {}
-          names.each do |name|
-            key = normalize_key(name, options)
-            entry = read_entry(key, options)
+          keys_to_names = Hash[names.map { |name| [normalize_key(name, options), name] }]
+          read_multi_entry(keys_to_names.keys, options).each_with_object({}) do |(key, entry), results|
```

Comment:
`each_with_object` is called just after another `each_with_object` in both `read_multi_entry` calls, which is to say, we're looping over the collection several times.

What's the performance implications of this? I'd love to see a benchmark so we know we don't have a regression here :smile: